tests/ui/crate_in_macro_def_allow.rs

```diff
@@ -0,0 +1,19 @@
+#![warn(clippy::crate_in_macro_def)]
+
+#[macro_use]
+mod intentional {
+    // For cases where use of `crate` is intentional, applying `allow` to the macro definition
+    // should suppress the lint.
+    #[allow(clippy::crate_in_macro_def)]
```

Comment:
This can go in the same test file as above or be removed, as you aren't having to do something specific to get it working

Could be worth noting in the description that the allow attribute has to go on the `macro_rules` itself, ie

```rust
#[allow(clippy::crate_in_macro_def)]
macro_rules! ok { ... crate::foo ... }

macro_rules! lints {
...
    // Still errors since the allow attribute is part of the macro token stream
     #[allow(clippy::crate_in_macro_def)]
    crate::foo
...
}
```

Reply:
> This can go in the same test file as above or be removed

My concern is that someone uses `crate` intentionally, they apply `allow` to the macro definition, and then a future change to Clippy breaks their uses of `allow`. E.g., it would be annoying to have to use `allow` at every call site. The test is meant to help prevent this possibility.